wallet/src/ui/app/redux/slices/txresults/index.ts

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import {
+    getTransactionDigest,
+    getTransactions,
+    getTransactionKindName,
+    getTransferCoinTransaction,
+    getExecutionStatusType,
+    getTotalGasUsed,
+} from '@mysten/sui.js';
+import {
+    createSlice,
+    createAsyncThunk,
+    createEntityAdapter,
+} from '@reduxjs/toolkit';
+
+import type {
+    TransactionEffectsResponse,
+    GetTxnDigestsResponse,
+    CertifiedTransaction,
+    TransactionKindName,
+    ExecutionStatusType,
+} from '@mysten/sui.js';
+import type { AppThunkConfig } from '_store/thunk-extras';
+
+type TxResultState = {
+    To?: string;
+    seq: number;
+    txId: string;
+    status: ExecutionStatusType;
+    txGas: number;
+    kind: TransactionKindName | undefined;
+    From: string;
+};
+
+const Txdapter = createEntityAdapter<TxResultState>({
+    selectId: ({ txId }) => txId,
+});
+
+interface TransactionManualState {
+    loading: boolean;
+    error: false | { code?: string; message?: string; name?: string };
+    latestTx: TxResultState[];
+}
+
+const initialState = Txdapter.getInitialState<TransactionManualState>({
+    loading: true,
+    latestTx: [],
+    error: false,
+});
+type TxResultByAddress = TxResultState[];
+
+const deduplicate = (results: [number, string][] | undefined) =>
+    results
+        ? results
+              .map((result) => result[1])
+              .filter((value, index, self) => self.indexOf(value) === index)
```

Comment:
Might be worth it to add a comment why we need this. Why do we get the same transaction id multiple times?